eclair-core/src/main/scala/fr/acinq/eclair/db/BackupHandler.scala

```diff
@@ -0,0 +1,46 @@
+package fr.acinq.eclair.db
+
+import java.io.File
+
+import akka.actor.{Actor, ActorLogging, Props}
+import fr.acinq.eclair.channel.ChannelPersisted
+
+
+/**
+  * README !
+  * This actor will synchronously make a backup of the database it was initialized with whenever it receives
+  * a ChannelPersisted event.
+  * To avoid piling up messages and entering an endless backup loop, it is supposed to be used with a bounded mailbox
+  * with a single item:
+  *
+  * backup-mailbox {
+  *   mailbox-type = "akka.dispatch.BoundedMailbox"
+  *   mailbox-capacity = 1
+  *   mailbox-push-timeout-time = 0
+  * }
+  *
+  * Messages that cannot be processed will be sent to dead letters
+  *
+  * @param databases database to backup
+  * @param wip work-in-progress file
+  * @param destination destination file
+  */
+class BackupHandler(databases: Databases, wip: File, destination: File) extends Actor with ActorLogging {
+
+  // we listen to ChannelPersisted events, which will trigger a backup
+  context.system.eventStream.subscribe(self, classOf[ChannelPersisted])
+
+  def receive = {
+    case persisted: ChannelPersisted =>
+      val start = System.currentTimeMillis()
+      databases.backup(wip)
+      val result = wip.renameTo(destination)
```

Comment:
```suggestion
      val result = tmpFile.renameTo(backupFile)
```

Reply:
From [javadoc](https://docs.oracle.com/javase/7/docs/api/java/io/File.html#renameTo(java.io.File)):

> Many aspects of the behavior of this method are inherently platform-dependent: The rename operation might not be able to move a file from one filesystem to another, **it might not be atomic**, and it might not succeed if a file with the destination abstract pathname already exists. The return value should always be checked to make sure that the rename operation was successful.